File: src/main/java/net/atos/client/zgw/zrc/model/zaakobjecten/ZaakobjectAdres.java
/*
 * SPDX-FileCopyrightText: 2023 Atos, 2025 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.client.zgw.zrc.model.zaakobjecten;

import java.net.URI;

import nl.info.client.zgw.zrc.model.generated.ObjectTypeEnum;


/**
 * ZaakobjectAdres
 */
public class ZaakobjectAdres extends ZaakobjectMetObjectIdentificatie<ObjectAdres> {

    /**
     * Constructor for JSONB deserialization
     */
    public ZaakobjectAdres() {
    }

    /**
     * Constructor with required attributes
     */
    public ZaakobjectAdres(final URI zaak, final URI bagobjectURI, final ObjectAdres adres) {
        super(zaak, bagobjectURI, ObjectTypeEnum.ADRES, adres);
    }

    @Override
    public String getWaarde() {
        return getObjectIdentificatie().getIdentificatie();
    }
}

File: src/main/kotlin/nl/info/client/zgw/zrc/model/zaakobjecten/ZaakobjectAdres.kt
/*
 * SPDX-FileCopyrightText: 2023, 2025, 2026 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.client.zgw.zrc.model.zaakobjecten

import nl.info.client.zgw.zrc.model.generated.ObjectTypeEnum
import java.net.URI

/**
 * ZaakobjectAdres
 */
class ZaakobjectAdres : ZaakobjectMetObjectIdentificatie<ObjectAdres> {
    /**
     * Constructor for JSONB deserialization
     */
    constructor() : super()

    /**
     * Constructor with required attributes
     */
    constructor(zaak: URI?, bagobjectURI: URI?, adres: ObjectAdres?) : super(zaak, bagobjectURI, ObjectTypeEnum.ADRES, adres)

    override val waarde: String?
        get() = objectIdentificatie?.identificatie
}
